src/gsy_e/models/market/two_sided.py

```diff
@@ -423,11 +437,39 @@ def validate_bid_offer_match(
                 f"{recommendation.bid_energy_rate}.")
         if market_offer.energy_rate > clearing_rate + FLOATING_POINT_TOLERANCE:
             raise InvalidBidOfferPairException(
-                f"Trade rate {clearing_rate} is higher than offer energy rate "
+                f"Trade rate {clearing_rate} is lower than offer energy rate "
                 f"{market_offer.energy_rate}.")
 
+        self._validate_matching_requirements(recommendation)
         self._validate_requirements_satisfied(recommendation)
 
+    @staticmethod
+    def _validate_matching_requirements(recommendation: BidOfferMatch) -> None:
+        """Validate a matching_requirement actually exists in the Bid/Offer object.
+
+        Raises:
+            InvalidBidOfferPairException: matching_requirement doesn't exist in the Bid/Offer
+            object.
+        """
+        if not recommendation.matching_requirements:
+            return
+
+        bid_matching_requirement = recommendation.matching_requirements.get("bid_requirement")
+        if bid_matching_requirement:
+            bid_requirements = recommendation.bid.get("requirements") or []
+            if bid_matching_requirement not in bid_requirements:
+                raise InvalidBidOfferPairException(
+                    f"Matching requirement {bid_matching_requirement} doesn't exist in the Bid"
+                    f" object.")
```

Comment:
Nit, same applies below 
```suggestion
                    " object.")
```